src/prime_rl/orchestrator/advantage.py

```
@@ -96,7 +98,21 @@ def compute_advantages(
     for group_rewards, group_lengths in zip(all_group_rewards, all_group_lengths):
         group_rewards_tensor = torch.tensor(group_rewards)
         group_lengths_tensor = torch.tensor(group_lengths)
-        group_advantages_tensor = compute_advantage(group_rewards_tensor, group_lengths_tensor)
+
+        if keep_n_shortest is not None:
+            k = min(max(keep_n_shortest, 0), group_rewards_tensor.numel())
```

could also fail loudly if the value is not in [0,G] imo, rather than doing these fallbacks but up to you